modules/oe_webtools_media/src/Plugin/media/Source/Webtools.php

```diff
@@ -100,16 +110,33 @@ public function getWidgetTypes() {
       'chart' => [
         'name' => $this->t('Chart'),
         'service' => 'charts',
+        'default_thumbnail' => 'charts-embed-no-bg.png',
       ],
       'map' => [
         'name' => $this->t('Map'),
         'service' => 'map',
+        'default_thumbnail' => 'maps-embed-no-bg.png',
       ],
       'social_feed' => [
         'name' => $this->t('Social feed'),
         'service' => 'smk',
+        'default_thumbnail' => 'twitter-embed-no-bg.png',
       ],
     ];
   }
 
+  /**
+   * Gets the thumbnail image URI based on widget type.
+   *
+   * @return string
+   *   URI of the thumbnail or NULL if there is no specific icon.
+   */
+  protected function getThumbnail(): ?string {
```

Comment:
Please also clean-up this method a bit. We don't need to check for the presence of the default thumbnail cause you hardcode them in the getWidgetTypes() method. So something like this:

```
  protected function getThumbnail(): ?string {
    $widget_type = $this->configuration['widget_type'];
    $icon_base = $this->configFactory->get('media.settings')->get('icon_base_uri');
    return $icon_base . '/' . $this->getWidgetTypes()[$widget_type]['default_thumbnail'];
  }
```